### DISCUSSION TITLE
Improving Person Detection (false positives)

### DISCUSSION THREAD

**Discussion by horspiste** [blakeblackshear/frigate#4759] [Ask A Question]
Hi! I wonder if there is something I could do to improve person detection? It sometimes detect trees, clouds shadows on the ground, or, as the attached picture shows, birds as persons. And other times it doesn't detect me. For now I have zoomed the camera to have a narrower field of view, less objects that can confuse the detection, but it is still happening and I'd like to monitor the wider, normal, field of view in front of the camera.

I have included another false detection, pretty high confidence, pretty large number of pixels, a snowblower this time.

Should I change my camera for one with a different resution or focal length?
Or do I have to wait for the Frigate+ online trainer and do my own training for my field of view?
Any suggestions?

Thanks!

![snapshot-1671548553 721607-pyss0q](https://user-images.githubusercontent.com/8324339/208721599-7903d175-5bd1-47be-8052-ae344861cb07.jpg)

![snapshot-1671275448 352312-54al1i](https://user-images.githubusercontent.com/8324339/208723526-e2f4f703-c94a-4be1-9232-19b37f321c77.jpg)

**Comment by NickM-27** [comment]
Frigate+ will offer models with better detection that doesn't have as many false positives. 

For now, there are many tools in frigate to narrow down false positives. Objects like the snow blower that are in the same place can be avoid using an object mask. It is imperative that you understand that an object mask is evaluated based on the bottom center of the bounding box. So a very small person mask at the bottom center area of that box will block it from being detected.

For objects like the bird you can set `min_area` filter for person to some value greater than 1254 which will avoid detect small objects like that as person.

**Reply by horspiste** [reply]
Thank you. The snow blower is mobile, it was clearing my parking space. I'll playing with the min area, since my camera sees far, it is usefull that it detects people on their way, but looking at my other proper detections I see that the smallest one was 1225. It would probably be ok to wait for people to be a bit closer to be detected.
